pkg/model/api/v1/datasource/sql/sql.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2025 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sql
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/go-sql-driver/mysql"
+	"time"
+)
+
+// Driver the SQL driver to use
+type Driver string
+
+const (
+	DriverMySQL      Driver = "mysql"
+	DriverPostgreSQL Driver = "postgres"
+)
+
+// SSLMode postgres ssl modes
+type SSLMode string
+
+const (
+	SSLModeDisable    SSLMode = "disable"
+	SSLModeAllow      SSLMode = "allow"
+	SSLModePreferable SSLMode = "prefer"
+	SSLModeRequire    SSLMode = "require"
+	SSLModeVerifyFull SSLMode = "verify-full"
+	SSLModeVerifyCA   SSLMode = "verify-ca"
+)
+
+type PostgresConfig struct {
+	// MaxConns is the maximum size of the pool
+	MaxConns int32 `json:"max_conns,omitempty" yaml:"max_conns,omitempty"`
```

Comment:
in the datamodel, we are following the Kubernetes convention, so every key should follow the camel case convention

For example:

```suggestion
MaxConns int32 `json:"maxConns,omitempty" yaml:"maxConns,omitempty"`
```

It's different in the configuration, we are following the snake_case because it helps to determinate the environment variable that can be used to override the configuration.